### ISSUE TITLE
Can onnxscript help me convert an intermediate format IR into the ONNX format?

### ISSUE THREAD

**Issue by sanbuphy** [microsoft/onnxscript#868]
Hello, I am currently working on something similar to PyTorch FX. What I need is the ability to load arbitrary model weights and then create an ONNX structure based on the optimized graph. I would like to add the model weights to the ONNX weights. Can you please tell me which API I can use to read a PyTorch weight dictionary and map it to the nodes created inside ONNX (similar to the original helper make node)? Furthermore, I would like to do this mapping in batches. Thank you!

**Comment by justinchuby** [comment]
Can you share more on how the ONNX model was created?

**Comment by sanbuphy** [comment]
> Can you share more on how the ONNX model was created? I imagine you would need a mapping information between the torch weights and the onnx weights first.

I have already implemented the mapping of the LeNet PyTorch model to an ONNX model. However, the method I used is quite cumbersome as I had to map the dictionary's key-value pairs one by one. I'm wondering if there is a better way to map the weights of the dictionary directly to a "make node" function, provided that ONNX has corresponding supported operators, instead of transferring and making nodes one by one.

![image](https://github.com/microsoft/onnxscript/assets/96160062/b4f5025c-4a03-458f-9479-3a99972b64f5)

**Comment by justinchuby** [comment]
I suspect you would be able to define a name mapping of the weights and the initializers, create the ONNX model, then insert the weights into the ONNX model using that mapping. Does that sound like something that will work?

**Comment by sanbuphy** [comment]
> I suspect you would be able to define a name mapping of the weights and the initializers, create the ONNX model, then insert the weights into the ONNX model using that mapping. Does that sound like something that will work?

Yes,  Does the method like `torch.onnx.export` implemented like this?

**Comment by justinchuby** [comment]
Not that I know of, but I think you will be able to load your weights first to the PyTorch model, then export?

**Comment by sanbuphy** [comment]
> Not that I know of, but I think you will be able to load your weights first to the PyTorch model, then export?

Actually , I want to write a function that can export model like the  'Pytorch model export'

**Comment by justinchuby** [comment]
I see, so you would like to create a converter. We are going to support procedural graph construction in onnxscript. So far there is one under torch_lib/graph_building.py that is specifically designed for PyTorch. For now I would suggest checking out https://github.com/Quantco/spox for your immediate use - https://spox.readthedocs.io/en/latest/manual/unstable.html#initializers initializers seems to be unstable as well.

**Comment by sanbuphy** [comment]
> I see, so you would like to create a converter. We are going to support procedural graph construction in onnxscript. So far there is one under torch_lib/graph_building.py that is specifically designed for PyTorch. For now I would suggest checking out https://github.com/Quantco/spox for your immediate use - https://spox.readthedocs.io/en/latest/manual/unstable.html#initializers initializers seems to be unstable as well.

Thank you so much! I will go there and take a look.
